app/controllers/state_file/questions/nj_homeowner_eligibility_controller.rb

```diff
@@ -0,0 +1,29 @@
+module StateFile
+  module Questions
+    class NjHomeownerEligibilityController < QuestionsController
+      include ReturnToReviewConcern
+
+      before_action -> { @filing_year = Rails.configuration.statefile_current_tax_year }
+
+      def self.show?(intake)
+        intake.household_rent_own == 'own'
+      end
+
+      def next_path
+        options = {}
+        options[:return_to_review] = params[:return_to_review] if params[:return_to_review].present?
+
+        is_hard_no = (current_intake.homeowner_home_subject_to_property_taxes == 'no' or (current_intake.homeowner_main_home_multi_unit == 'yes' and current_intake.homeowner_main_home_multi_unit_max_four_one_commercial == 'no'))
+        is_soft_no = (current_intake.homeowner_more_than_one_main_home_in_nj == 'yes' or current_intake.homeowner_shared_ownership_not_spouse == 'yes' or (current_intake.homeowner_main_home_multi_unit == 'yes' and current_intake.homeowner_main_home_multi_unit_max_four_one_commercial == 'yes'))
```

Comment:
This logic gets repeated in 3 different components. I want to externalize it so that it's shared, but I'm not sure _where_ the best place for that to live is. I think a question for CFA engineers

Reply:
I would move it into a method (that takes an intake) in a new module in app/helpers (possibly in a new state_file subdirectory and named something like `nj_homeowner_helper.rb`), and include that module in each class where you currently have this logic.